exporter/exporterhelper/common.go

```diff
@@ -102,19 +101,19 @@ func fromOptions(options []Option) *baseSettings {
 // Option apply changes to baseSettings.
 type Option func(*baseSettings)
 
-// WithShutdown overrides the default Shutdown function for an exporter.
+// WithStart overrides the default Start function for an exporter.
 // The default shutdown function does nothing and always returns nil.
-func WithShutdown(shutdown componenthelper.Shutdown) Option {
+func WithStart(start componenthelper.Start) Option {
```

Comment:
Rather than needing to duplicate every `componenthelper.Option` here, might it be better to have something like this:

```go
func WithComponentOptions(opts componenthelper.Option...) Option {
  return func( o *baseSettings) {
    o.componentOptions = append(o.componentOptions, opts...)
  }
}
```

This could then be invoked like so:

```go
exp := newBaseExporter(
  cfg,
  logger,
  WithComponentOptions(
    componenthelper.WithStart(start),
    componenthelper.WithShutdown(shutdown),
  ),
)
```

This would simplify the path forward for adding more `componenthelper.Option`s

Reading on further, I see that these `Option`s were already in use so changing to a more general approach would then impact more call sites and break compatibility.  I guess it then becomes a question of whether it's anticipated that there will be more `componenthelper.Option` implementations and what the value of avoiding duplicating them in other packages is.  I don't think I have enough understanding of the codebase as a whole to offer an opinion on that subject at this time.

Reply:
https://github.com/open-telemetry/opentelemetry-collector/issues/2787
https://github.com/open-telemetry/opentelemetry-collector/issues/2788

Reply:
Filed issues, the exporter/processor helper are for phase2